packages/@aws-cdk/core/lib/bundling.ts

```diff
@@ -21,6 +21,21 @@ export class DockerBuildSecret {
   public static fromSrc(src: string): string {
     return `src=${src}`;
   }
+
+  /**
+   * A Docker build secret from the build environment
+   * @param environmentVariable The environment variable name
+   * @returns The latter half required for `--secret`
+   */
+  public static fromEnvironment(environmentVariable: string): string {
+    const value = process.env[environmentVariable];
+
+    if (value) {
+      return `env=${environmentVariable}`;
+    } else {
+      throw new Error(`Environment variable ${environmentVariable} is not set`);
```

Comment:
At what point does this error message get surfaced? Will it be in a context that this message will be clear enough to a user?